components/kyma-environment-broker/internal/process/input/input.go

```diff
@@ -477,9 +478,6 @@ func (r *RuntimeInput) applyProvisioningParametersForUpgradeShoot() error {
 		}
 	}
 
-	// use autoscaler value in provisioningParameters if it is not nil
-	updateInt(r.upgradeShootInput.GardenerConfig.AutoScalerMin, r.provisioningParameters.Parameters.AutoScalerMin)
-	updateInt(r.upgradeShootInput.GardenerConfig.AutoScalerMax, r.provisioningParameters.Parameters.AutoScalerMax)
```

Comment:
I think we should allow to user to change the autoscaler params

Reply:
The user can change the autoscalert params, the update processing always use the input from the updating parameters:
https://github.com/kyma-project/control-plane/blob/main/components/kyma-environment-broker/internal/process/update/upgrade_shoot_step.go#L105-L112

The removal of this code (setting input autoscaler params from provisioning params) is because provisioner upgradeShoot API works as a PATCH API, i.e we only need to provide autoscaler params during provisioning, and at any user update (lines above in the process step). We don't need to provide user provided autoscaler params at every upgradeShoot input (used by all cluster upgrade and instance update operations).